readme.md

```diff
@@ -0,0 +1,63 @@
+# ACME® Corporation Stores
+
+## Usage
+
+### Run the app
+
+1. Run `docker-compose up` at the root to start the webapp, backend, and data importer.
+2. Wait a few seconds because the loader takes a while to import stores to the local database
+3. [Go to localhost:8080](http://localhost:8080)
+
+### Run the tests
+
+- Run `./test.sh` so that the whole suite of tests is run.
+- Run `npm t` to run the UI tests. Make sure the app has some data first by running `docker-compose up` first.
+
+## User stories
+
+- List stores ✅
+  - paginate list of stores ✅
+  - fetch data from APIs ✅
+- View store details ✅
+  - show basic info ✅
+  - view extra fields ✅
+  - view seasons ✅
+- Search stores by name ✅
+- Edit store name ✅
+- ✨ Download all stores as CSV ✅
+- ✨ Scheduled auto-importer ✅
+
+# Design decisions
+
+## Principles
+
+- Lean approach: I tried to do the minimum needed to deliver user value and reduce over-engineering:
+  the least libraries, the minimum assumptions, the least amount of code, a small set of technologies. This leaves
+  options open for a bigger amount of time.
+- TDD: I followed test-driven development most of the time: in fact, that's my documentation. I don't believe in
+  documentation through code comments since no one maintains them. Worse, people will rely on them to explain bad code
+  rather than improving it.
+- Clean architecture: I followed the principles of the clean architecture: a core domain and segregated adapters around
+  it. Dependencies point inwards.
+- Full-stack: I don't like doing frontend separated from backend. The only way to deliver quick user value is to deliver
```

Comment:
are you referring to having the frontend separated from the backend? if so, I'd say unfortunately there's not a clear moment to do it; it's the sum of multiple factors:
- if the FE becomes richer (e.g. a SPA) and its development becomes cumbersome the way it is
- if the teams are separated by FE/BE (which I don't recommend)
- if the FE needs another set of tools like Babel, minification, bundling, or any other kind of processing that can't be easily done statically.
- if it needs to be deployed separately for performance reasons

These decisions have implications and a price to pay on the development efforts. Therefore, these factors need frequent evaluation by the team so that the moment is decided if/when neeed.

Reply:
another breaking point would be the FE needing its own BE (e.g. for authentication, proxying, etc)